core/instance.go

```diff
@@ -269,52 +273,222 @@ func (i *instance) getCheapestCompatibleSpotInstanceType(allowedList []string, d
 			i.isAllowed(candidate.instanceType, allowedList, disallowedList) {
 			bestPrice = candidatePrice
 			chosenSpotType = candidate.instanceType
+			cheapest = candidate
 			debug.Println("Best option is now: ", chosenSpotType, " at ", bestPrice)
 		} else if chosenSpotType != "" {
 			debug.Println("Current best option: ", chosenSpotType, " at ", bestPrice)
 		}
 	}
 	if chosenSpotType != "" {
 		debug.Println("Cheapest compatible spot instance found: ", chosenSpotType)
-		return chosenSpotType, nil
+		return cheapest, nil
 	}
-	return chosenSpotType, fmt.Errorf("No cheaper spot instance types could be found")
+	return cheapest, fmt.Errorf("No cheaper spot instance types could be found")
```

Comment:
The first thing I wondered is: "Is cheapest nil?"
I suppose that it is, but I feel that it would be better to remove the doubt, by returning explicitly nil.

Even if it isn't the code later on, check the error returned value, and doesn't use cheapest, so it is unnecessary to return it.

Reply:
Thanks, I'll fix this soon.